i6_models/assemblies/conformer/conformer.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import annotations
+import torch
+from torch import nn
+
+# TODO probably used for SpecAugment
+from torchaudio.transforms import TimeMasking, TimeStretch, FrequencyMasking
+from dataclasses import dataclass
+
+from i6_models.config import ModelConfiguration
+
+from i6_models.parts.conformer.convolution import ConformerConvolutionV1
+from i6_models.parts.conformer.feedforward import ConformerPositionwiseFeedForwardV1
+from i6_models.parts.conformer.mhsa import ConformerMHSAV1
+
+
+@dataclass
+class ConformerFeedForwardV1Config(ModelConfiguration):
+    dim: int
+
+
+@dataclass
+class ConformerMHSAV1Config(ModelConfiguration):
+    pass
+
+
+@dataclass
+class ConformerConvV1Config(ModelConfiguration):
+    pass
+
+
+@dataclass
+class ConformerBlockV1Config(ModelConfiguration):
+    # hyperparameters
+    norm_dim: int
+
+    # nested configurations
+    ff_cfg: ConformerFeedForwardV1Config  # Configuration for ConformerPositionwiseFeedForwardV1
+    mhsa_cfg: ConformerMHSAV1Config  # Configuration for ConformerMHSAV1
+    conv_cfg: ConformerConvV1Config  # Configuration for ConformerConvolutionV1
+
+
+@dataclass
+class ConformerFrontendV1Config(ModelConfiguration):
+
+    feature_dim: int
+    hidden_dim: int
+    dropout: float
+
+    # TODO: Maybe put this in own config?
+    conv_stride: int
+    conv_kernel: int
+    conv_stride: int
+    conv_padding: int
+
+    spec_aug_cfg: ModelConfiguration  # TODO
+
+
+@dataclass
+class ConformerV1Config(ModelConfiguration):
+    # hyperparameters
+    num_layers: int
+
+    # nested configurations
+    front_cfg: ConformerFrontendV1Config
+    block_cfg: ConformerBlockV1Config
+
+
+class ConformerBlockV1(nn.Module):
+    """
+    Conformer block module
+    """
+
+    def __init__(self, cfg: ConformerBlockV1Config):
+        """
+        :param cfg: Conformer Block Configuration with subunits for the different Conformer parts
+        """
+        super().__init__()
+
+        self.ff_config = cfg.ff_cfg
+        self.mhsa_config = cfg.mhsa_cfg
+        self.conv_config = cfg.conv_cfg
```

Comment:
Do we really need these as class attributes?

Reply:
we could also either keep the whole `cfg` as class attribute or none of it. I don't have a strong opinion on this.

Reply:
I would say not to set attributes that are never used anywhere else.

Reply:
so not even `self.cfg` then?

Reply:
Tbh. I just thought it would be nice to have it available without further thinking, but I will remove it for now and we can add it later if required.